instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/utils/NetPeerUtils.java

```diff
@@ -17,13 +18,17 @@
 
 public final class NetPeerUtils {
 
-  private static final Map<String, String> ENDPOINT_PEER_SERVICE_MAPPING =
-      Collections.unmodifiableMap(
-          Config.get().getMapProperty("otel.endpoint.peer.service.mapping"));
+  public static final NetPeerUtils INSTANCE = new NetPeerUtils(Config.get());
 
-  private NetPeerUtils() {}
+  private final Map<String, String> endpointPeerServiceMapping;
 
-  public static void setNetPeer(Span span, @Nullable InetSocketAddress remoteConnection) {
+  // visible for testing
+  NetPeerUtils(Config config) {
+    this.endpointPeerServiceMapping =
+        Collections.unmodifiableMap(config.getMapProperty("otel.endpoint.peer.service.mapping"));
+  }
+
+  public void setNetPeer(Span span, @Nullable InetSocketAddress remoteConnection) {
```

Comment:
any thoughts/preferences on this approach to making this class more testable?

Reply:
Dependency injection like this is a standard approach I think 👍